assigning bug with math.abs

### Describe the bug

I wanted to do this in one line so I don't have to use a temp variable.
```v
x, y = y, math.abs(x)
```

Tested it with  
```v
x, y = y, x
```
I thought it would work with abs too but it doesn't compile

### Expected Behavior

A compiled program printing : 

-2 | 1

### Current Behavior

C:/Users/lopottie/AppData/Local/Temp/v_0/abs_bug.8046743995589138909.tmp.c:499: warning: WINVER redefined
C:/Users/lopottie/AppData/Local/Temp/v_0/abs_bug.8046743995589138909.tmp.c:7203: warning: implicit declaration of function 'tcc_backtrace'    
C:/Users/lopottie/AppData/Local/Temp/v_0/abs_bug.8046743995589138909.tmp.c:7951: warning: assignment from incompatible pointer type
C:/Users/lopottie/AppData/Local/Temp/v_0/abs_bug.8046743995589138909.tmp.c:7951: warning: assignment makes pointer from integer without a cast
C:/Users/lopottie/AppData/Local/Temp/v_0/abs_bug.8046743995589138909.tmp.c:7951: warning: cast between pointer and integer of different size  
C:/Users/lopottie/AppData/Local/Temp/v_0/abs_bug.8046743995589138909.tmp.c:15268: warning: implicit declaration of function 'math__abs'       
tcc: error: undefined symbol 'math__abs'
builder error:

C error. This should never happen.

### Reproduction Steps

```v
import math

fn main() {
	mut x := -1
	mut y := -2
	x, y = y, math.abs(x)
	println("${x} | ${y}")
}
```

### Possible Solution

_No response_

### Additional Information/Context

I tried putting math.abs in a separate line and it compiles.
Builder fail only when I put everything on the same line.

### V version

V 0.3.3 f598bdd

### Environment details (OS name and version, etc.)

V full version: V 0.3.3 524f7c3.f598bbd
OS: windows, Microsoft Windows 10 Entreprise v19044 64 bits
Processor: 8 cpus, 64bit, little endian,

getwd: C:\Users\lopottie\Documents\GitHub\v
vexe: C:\Users\lopottie\Documents\GitHub\v\v.exe
vexe mtime: 2023-04-24 11:59:44

vroot: OK, value: C:\Users\lopottie\Documents\GitHub\v
VMODULES: OK, value: C:\Users\lopottie\.vmodules
VTMP: OK, value: C:\Users\lopottie\AppData\Local\Temp\v_0

Git version: git version 2.39.2.windows.1
Git vroot status: weekly.2023.15-57-gf598bbde
.git/config present: true

CC version: Error: exec failed (CreateProcess) with code 2: Le fichier spécifié est introuvable.
 cmd: cc --version
thirdparty/tcc status: thirdparty-windows-amd64 e90c2620

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -985,7 +985,10 @@ fn (mut g Gen) gen_cross_tmp_variable(left []ast.Expr, val ast.Expr) {
 			g.write(')')
 		}
 		ast.CallExpr {
-			fn_name := val.name.replace('.', '__')
+			mut fn_name := val.name.replace('.', '__')
+			if val.concrete_types.len > 0 {
+				fn_name = g.generic_fn_name(val.concrete_types, fn_name)
+			}
 			g.write('${fn_name}(')
 			for i, arg in val.args {
 				g.gen_cross_tmp_variable(left, arg.expr)

--- a/vlib/v/tests/cross_assign_with_generic_fn_call_test.v
+++ b/vlib/v/tests/cross_assign_with_generic_fn_call_test.v
@@ -1,0 +1,10 @@
+import math
+
+fn test_cross_assign_with_generic_fn_call() {
+	mut x := -1
+	mut y := -2
+	x, y = y, math.abs(x)
+	println('${x} | ${y}')
+	assert x == -2
+	assert y == 1
+}

